drivers/storage/portworx/portworx_test.go

```diff
@@ -6621,16 +6619,22 @@ func TestStorageClusterDefaultsForTelemetry(t *testing.T) {
 		},
 		Spec: corev1.StorageClusterSpec{
 			Image: "px/image:2.7.0",
+			Monitoring: &corev1.MonitoringSpec{
+				Telemetry: &corev1.TelemetrySpec{
+					Enabled: true,
+				},
+			},
 		},
 	}
 
-	// Don't enable telemetry by default < 2.8.0
+	// Disable telemetry for px version < 2.8.0
 	driver.SetDefaultsOnStorageCluster(cluster)
-	require.Empty(t, cluster.Spec.Monitoring)
+	require.False(t, cluster.Spec.Monitoring.Telemetry.Enabled)
 	require.Empty(t, cluster.Status.DesiredImages.Telemetry)
 
-	// Enable telemetry by default for 2.8.0
+	// Allow telemetry to be enabled for 2.8.0
 	cluster.Spec.Image = "px/image:2.8.0"
+	cluster.Spec.Monitoring.Telemetry.Enabled = true
 	driver.SetDefaultsOnStorageCluster(cluster)
 	require.NotEmpty(t, cluster.Spec.Monitoring) // telemetry is under monitoring
 	require.NotEmpty(t, cluster.Spec.Monitoring.Telemetry)
```

Comment:
should we change this?
```suggestion
      require.True(t, cluster.Spec.Monitoring.Telemetry.Enabled)
```

Reply:
We are doing that check on the next line.

Reply:
Ah right. Didn't see it in the review mode.